api/ngrok/v1alpha1/agentendpoint_types.go

```diff
@@ -0,0 +1,169 @@
+/*
+MIT License
+
+Copyright (c) 2024 ngrok, Inc.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+package v1alpha1
+
+import (
+	"encoding/json"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// AgentEndpoint is the Schema for the agentendpoints API
+//
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:categories="networking";"ngrok"
+// +kubebuilder:resource:shortName=aep
+// +kubebuilder:printcolumn:name="URL",type="string",JSONPath=".status.assignedURL"
+// +kubebuilder:printcolumn:name="Traffic Policy",type="string",JSONPath=".status.trafficPolicy"
```

Comment:
We might want to set a priority on this if `.status.trafficPolicy` is the full traffic policy string so it only shows up with `-o wide`

Reply:
It would only display `"inline"`, `"none"`, or `"<name>"` to indicate where the traffic policy comes from, not including the whole raw string if it is inlined. Just to help give users debugging/editing a quick heads-up on where the policy comes from without needing to fully inspect/edit the AgentEndpoint.

Reply:
Removed for now since we aren't going to support the status yet